overlord/snapshotstate/snapshotstate.go

```diff
@@ -69,6 +73,70 @@ func allActiveSnapNames(st *state.State) ([]string, error) {
 	return names, nil
 }
 
+func automaticSnapshotExpiration(st *state.State) time.Duration {
+	// TODO: get from config
+	return DefaultAutomaticSnapshotExpiration
+}
+
+// saveExpiration saves expiration date of the given snapshot set in the state.
+// The state needs to be locked by the caller.
+func saveExpiration(st *state.State, setID uint64, expiryTime time.Time) error {
+	// note, expirations actually map to time.Time, use string here as we don't need parsed times here.
+	var expirations map[uint64]string
+	err := st.Get("snapshots-expiry", &expirations)
```

Comment:
Nitpick: could you make `snapshots-expiry` a const string defined at file level?

Reply:
we actually tend not to do that (see "snaps"), also I'm not sure I'm happy with that as a toplevel entry in data, I would leave this alone until I get to review this

Reply:
This is typo material, why are we not using a constant for that?

Reply:
Because the string, if affordable, is friendlier to a reader, no wondering what it is really, not wondering whether is a variable.  We have tests to catch typos,  the tests shouldn't use the constant or the same constant anyway. If we can let through typos everywhere then the constant will not help us either.

We use ...Get(string literal...)  all over overlord (because of those reasons implicitly as far as I'm aware).

Reply:
While I don't disagree that is what we do I don't see the reason to use a literal so many times. Note that I suggested to use a constant, not a variable. While I don't think I'm convincing you on this case I still believe it's good practice to do use a constant rather than a number of literals.